app/experimenter/nimbus-ui/schema.graphql

```diff
@@ -491,100 +271,243 @@ type NimbusFeatureConfigType {
   application: NimbusExperimentApplicationEnum
   ownerEmail: String
   schema: String
-  readOnly: Boolean!
   setsPrefs: Boolean
-  nimbusexperimentSet: [NimbusExperimentType!]!
-  nimbusbranchfeaturevalueSet: [NimbusBranchFeatureValueType!]!
 }
 
-enum NimbusIsolationGroupApplication {
-  FIREFOX_DESKTOP
-  FENIX
-  IOS
-  FOCUS_ANDROID
-  KLAR_ANDROID
-  FOCUS_IOS
-  KLAR_IOS
-}
-
-type NimbusIsolationGroupType {
-  application: NimbusIsolationGroupApplication!
+type NimbusBranchType {
+  id: Int
   name: String!
-  instance: Int!
-  total: Int!
-  bucketRanges: [NimbusBucketRangeType!]!
+  slug: String!
+  description: String!
+  ratio: Int!
+  featureValues: [NimbusBranchFeatureValueType]
+  screenshots: [NimbusBranchScreenshotType!]
+  featureEnabled: Boolean!
+  featureValue: String
 }
 
-type NimbusLabelValueType {
-  label: String
+type NimbusBranchFeatureValueType {
+  id: Int
+  featureConfig: NimbusFeatureConfigType
+  enabled: Boolean
   value: String
 }
 
-type NimbusLanguageType {
+type NimbusBranchScreenshotType {
   id: Int
-  code: String
-  name: String
-  nimbusexperimentSet: [NimbusExperimentType!]!
+  image: String
+  description: String
 }
 
-type NimbusLocaleType {
-  id: Int
-  code: String
-  name: String
-  nimbusexperimentSet: [NimbusExperimentType!]!
+enum NimbusExperimentConclusionRecommendationEnum {
+  RERUN
+  GRADUATE
+  CHANGE_COURSE
+  STOP
+  FOLLOWUP
 }
 
-type NimbusOutcomeMetricType {
-  slug: String
-  friendlyName: String
-  description: String
+type NimbusDocumentationLinkType {
+  title: ExperimentsNimbusDocumentationLinkTitleChoices!
+  link: String!
 }
 
-type NimbusOutcomeType {
-  friendlyName: String
-  slug: String
-  application: NimbusExperimentApplicationEnum
-  description: String
-  isDefault: Boolean
-  metrics: [NimbusOutcomeMetricType]
+"""An enumeration."""
+enum ExperimentsNimbusDocumentationLinkTitleChoices {
+  """Data Science Jira Ticket"""
+  DS_JIRA
+
+  """Experiment Design Document"""
+  DESIGN_DOC
+
+  """Engineering Ticket (Bugzilla/Jira/GitHub)"""
+  ENG_TICKET
 }
 
+"""
+The `DateTime` scalar type represents a DateTime
+value as specified by
+[iso8601](https://en.wikipedia.org/wiki/ISO_8601).
+"""
+scalar DateTime
+
 type NimbusReviewType {
   message: ObjectField
   warnings: ObjectField
   ready: Boolean
 }
 
+"""Utilized to serialize out Serializer errors"""
+scalar ObjectField
+
+type NimbusChangeLogType {
+  changedOn: DateTime!
+  changedBy: NimbusUserType!
+  oldStatus: ExperimentsNimbusChangeLogOldStatusChoices
+  oldStatusNext: ExperimentsNimbusChangeLogOldStatusNextChoices
+  message: String
+}
```

Comment:
In general, if something is removed but tests still pass, then either you're missing tests or that code wasn't being used.  In this specific case, we used to define the gql types by saying 'this is the model, so automatically serialize all of its fields, but exclude these specific fields' so lots of fields were being exposed that were never touched by anything.  That's one of the nice things about the gql generated ts types, if you remove it from the backend but it's referenced from the frontend, you'll know.  

So I switched all the gql types to explicitly declare which fields to serialize, since that just generally makes it easier to reason about, so now all the fields that were being invisibly auto serialized are gone, and since all the type checking and tests pass, we know nothing was consuming them.

Reply:
Beautiful! It's great to know those weren't being used